nixos/modules/system/etc/setup-etc.pl

```diff
@@ -108,6 +108,8 @@ sub link {
             my $uid = read_file("$_.uid"); chomp $uid;
             my $gid = read_file("$_.gid"); chomp $gid;
             copy "$static/$fn", "$target.tmp" or warn;
+            $uid = getpwnam $uid unless $uid =~ /^\d+$/;
+            $gid = getgrnam $gid unless $gid =~ /^\d+$/;
```

Comment:
```chown``` (command line utility) mixes them as well.
It is only Perl limitation to accept only ids which leaked to NixOS.
Were that script written in Bash instead of Perl, we'd have mixed behaviour already